lib/resty/iprepd.lua

```diff
@@ -191,17 +199,28 @@ end
 
 function _M.audit_log(self, ip)
   if self.audit_blocked_requests == 1 then
-    ngx.req.read_body()
-    local headers = ngx.req.get_headers()
-    local request_headers_all = ""
-    for k,v in pairs(headers) do
-      request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+    local should_audit_request = 0
+    for i, uri in ipairs(self.audit_uri_list) do
+      if string.match(uri, ngx.var.uri) then
+        should_audit_request = 1
+        break;
+      end
     end
-    local data = ngx.req.get_body_data()
-    if not data then
-      data = ngx.req.get_body_file()
+    if should_audit_request == 1 then
+      ngx.req.read_body()
+      local request_headers_all = ""
+      if self.audit_include_headers == 1 then
+        local headers = ngx.req.get_headers()
+        for k,v in pairs(headers) do
+          request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+        end
+      end
+      local data = ngx.req.get_body_data()
+      if not data then
+        data = ngx.req.get_body_file()
```

Comment:
Is this always guaranteed to succeed? Do we need to check `data` again here and error out (or perhaps just default it to a string indicating it could not be read)?